### PR TITLE
Set LocalCUDACluster n_workers equal to the length of CUDA_VISIBLE_DE…

### PR DESCRIPTION
**Pull request by mrocklin**
…VICES

### CODE REVIEW

File: dask_cuda/local_cuda_cluster.py [hunk 1]
```diff
@@ -54,13 +54,13 @@ def __init__(
...
     ):
-        if n_workers is None:
-            n_workers = get_n_gpus()
         if CUDA_VISIBLE_DEVICES is None:
             CUDA_VISIBLE_DEVICES = cuda_visible_devices(0)
         if isinstance(CUDA_VISIBLE_DEVICES, str):
             CUDA_VISIBLE_DEVICES = CUDA_VISIBLE_DEVICES.split(",")
         CUDA_VISIBLE_DEVICES = list(map(int, CUDA_VISIBLE_DEVICES))
+        if n_workers is None:
+            n_workers = len(CUDA_VISIBLE_DEVICES)
```
Comment (pentschev) [inline dask_cuda/local_cuda_cluster.py]: What if the user doesn't pass `CUDA_VISIBLE_DEVICES`?

Comment (mrocklin) [reply dask_cuda/local_cuda_cluster.py]: I believe that it is handled with the set of checks above just above.

Comment (pentschev) [reply dask_cuda/local_cuda_cluster.py]: You're right. The `cuda_visible_devices(0)` got me confused, sorry for the false alarm.

**Comment by pentschev** [comment]
rerun tests

**Comment by pentschev** [comment]
rerun tests

**Comment by pentschev** [comment]
Finally this passes, I'm not sure what exactly is the problem of using `LocalCUDACluster` on a sync test, but it was affecting the tests that ran after somehow. Making it async as the other `LocalCUDACluster` tests seems to solve the issue. I don't know whether this is caused by some issue in `LocalCUDACluster` or not, but if someone has an idea, please let me know.

I'm going ahead and merging this before CI changes its mind.

**Pull request merged**